packages/flutter/lib/src/widgets/app.dart

```diff
@@ -422,6 +461,34 @@ class WidgetsApp extends StatefulWidget {
   /// {@endtemplate}
   final Iterable<LocalizationsDelegate<dynamic>> localizationsDelegates;
 
+  /// {@template flutter.widgets.widgetsApp.localeListResolutionCallback}
+  /// This callback is responsible for choosing the app's locale
+  /// when the app is started, and when the user changes the
+  /// device's locale.
+  ///
+  /// The returned value becomes the locale of this app's [Localizations]
+  /// widget. The callback's `locales` parameter is the device's preferred
+  /// locales list when the app started, or the device locales the user selected
+  /// after the app was started. The callback's `supportedLocales` parameter is
+  /// just the value [supportedLocales].
+  ///
+  /// If the callback is null or if it returns null then the resolved locale is:
+  ///
+  /// - The value returned by [localeResolutionCallback] if the callback is not null.
+  /// - The callback's `locale` parameter if it's equal to a supported locale.
+  /// - The first supported locale with the same [Locale.languageCode] and
+  ///   [Locale.scriptCode] as the callback's `locale` parameter.
+  /// - The first supported locale with the same [Locale.languageCode] as the
+  ///   callback's `locale` parameter.
+  /// - The first locale in [supportedLocales].
```

Comment:
this may be clearer as a numbered list with number 1 being the return value of this callback if it's not null.